### ISSUE TITLE
Barcode demultiplexing date

### ISSUE THREAD

**Issue by maximum411** [nanoporetech/dorado#257]
Is there an estimate for when barcode demultiplexing will be released?

I have been finding it a bit tricky to demultiplex Dorado basecalled data with Guppy- it didn't seem to work properly with the bam file output by Dorado (it was a single large BAM, so maybe that was why). I converted to FASTQ with Samtools, retained tags, and split into multiple smaller FASTQs, and was able to demultiplex using Gupppy barcoder, but after this the FASTQ tags appeared to cause problems with alignment and conversion back to BAM, so I ended up having to drop methylation tags in order to proceed with alignment.

**Comment by tijyojwad** [comment]
Hi @maximum411 - thanks for your interest in the feature! We're actively working on it and is our highest priority. Currently planned as part of out next major release.

**Comment by maximum411** [comment]
Great! Looking forward to it.

**Comment by HannahBenisty** [comment]
Hi, 

I am currently in the same process. When stereo basecalling with Dorado, is there a way to obtain also the fastq? Is --emit fastq working?

Thanks!

**Comment by tijyojwad** [comment]
Hi @HannahBenisty - yes `--emit-fastq` option is available in `dorado duplex` as well. if it's not working, please give us a holler

**Comment by HannahBenisty** [comment]
Now the basecalling is running and the bam file is getting updated, but no fastq file is created.

**Comment by JWDebler** [comment]
To extract simplex and duplex reads from your duplex.bam run this:

```
samtools view -@10 -ddx:1 -O fastq duplex.bam | gzip -9 > duplex.fastq.gz 
samtools view -@10 -ddx:0 -O fastq duplex.bam | gzip -9 > simplex.fastq.gz
```

**Comment by tijyojwad** [comment]
> Now the basecalling is running and the bam file is getting updated, but no fastq file is created.

@HannahBenisty we don't generate BAM and FASTQ simultaneously. dorado can only generate one file type at a time. If `--emit-fastq` is enabled then dorado write fastq to `stdout`. Default is BAM.

**Comment by ethan-mcq** [comment]
@tijyojwad as a feature, it would be nice to include a parameter to accept a bed file or similar with custom barcodes when multiplexing is released!!

**Comment by Ge0rges** [comment]
Hello, are there any updates on when the next major release is planned for? Thank you

**Comment by susie-ont** [comment]
Hi @Ge0rges ! We are planning the next major release by the end of September. Thanks for your interest!

**Comment by ethan-mcq** [comment]
@tijyojwad Thanks so much! I know a lot of people who would be grateful for that as they don't use the normal Nanopore barcodes in their workflows.

**Comment by tijyojwad** [comment]
Hello @Ge0rges @maximum411 - we just released dorado [v0.4.0](https://github.com/nanoporetech/dorado/releases/tag/v0.4.0) with barcode demultiplexing and trimming.

@ethan-mcq There's no support for custom barcodes yet, but we are working on adding it to a future release.
